### PR TITLE
Jv change executable file path

### PR DESCRIPTION
**Pull request by JoukoVirtanen**
## Description

Integration test where argv[0] is changed after creating a listening endpoint. This makes it so that `readlink exe` and cmdline give different executable file paths.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Updated documentation accordingly

**Automated testing**
  - [ ] Added unit tests
  - [ ] Added integration tests
  - [ ] Added regression tests

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why you did not do so. (Valid reasons include "CI is sufficient" or "No testable changes")
In addition to reviewing your code, reviewers **must** also review your testing instructions, and make sure they are sufficient.

For more details, ref the [Confluence page](https://stack-rox.atlassian.net/wiki/spaces/StackRox/pages/855998488/Proposal+Explicitly+List+Testing+Steps+on+PRs) about this section.

### CODE REVIEW

File: integration-tests/suites/change_process_name.go [hunk 1]
```diff
@@ -22,35 +27,42 @@ func (s *ChangeProcessNameTestSuite) SetupSuite() {
...
-	s.serverContainer = common.ContainerShortID(containerID)
+	s.container = common.ContainerShortID(containerID)
 
 	time.Sleep(20 * time.Second)
 }
 
 func (s *ChangeProcessNameTestSuite) TearDownSuite() {
-	s.cleanupContainer([]string{"change-process-name", "collector"})
+	s.StopCollector()
+	s.cleanupContainer([]string{"collector"})
```
Comment (JoukoVirtanen) [inline integration-tests/suites/change_process_name.go:41]: Not needed

File: integration-tests/container/processes-listening-on-ports/listening-endpoint-child-process-no-fork.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  int sockfd;
+  bool is_parent = true;
+
+  // Check if sockfd is provided as a command line argument
+  if (argc == 2) {
+    sockfd = atoi(argv[1]);
+    is_parent = false;
+  } else {
+    // Create a socket if sockfd is not provided as an argument
+    sockfd = socket(AF_INET, SOCK_STREAM, 0);
```
Comment (erthalion) [inline integration-tests/container/processes-listening-on-ports/listening-endpoint-child-process-no-fork.c:20]: This part about listening on a port seems to be repeated between the tests, what about reusing it?

**Comment by erthalion** [comment]
Just out of curiosity, what are the test case expectations exactly, why this combination of fork/exec/listen/filepath? Could it be written somewhere in the plain text?

File: integration-tests/integration_test.go [hunk 3]
```diff
@@ -123,8 +123,169 @@ func TestDuplicateEndpoints(t *testing.T) {
...
+					ProcessName:         "change-executab",
+					ProcessExecFilePath: "/change-executable-file-path",
+					ProcessArgs:         "",
+				},
+			},
+		},
+		ExpectedProcesses: []types.ProcessInfo{
+			{
+				Name:    "change-executab",
+				ExePath: "/change-executable-file-path",
```
[the remark this answers is not in the document]
Comment (JoukoVirtanen) [reply integration-tests/integration_test.go:180]: Collector doesn't see it. Neither does lsof. It is seen by netstat, `ps -aux`, and readlink.

File: integration-tests/container/processes-listening-on-ports/listening-endpoint-child-process-no-fork.c (hunk 2 above)
Comment (JoukoVirtanen) [reply integration-tests/container/processes-listening-on-ports/listening-endpoint-child-process-no-fork.c:23]: Good idea. Will do.

**Pull request merged**